draco/analysis/transform.py

```diff
@@ -200,15 +209,8 @@ def pack_product_array(arr):
         bt_freq = ss.index_map['freq'][freq_ind]
 
         # Construct the equivalent prod and stack index_map for the telescope instance
-        bt_prod = np.array([(fi, fj) for fi in range(self.telescope.nfeed)
-                                     for fj in range(fi, self.telescope.nfeed)],
-                                     dtype=[('input_a', '<u2'), ('input_b', '<u2')])
-
-        bt_stack = np.array([(tools.cmap(upp[0], upp[1], self.telescope.nfeed), 0)
-                             if upp[0] <= upp[1] else
-                             (tools.cmap(upp[1], upp[0], self.telescope.nfeed), 1)
-                             for upp in self.telescope.uniquepairs],
-                             dtype=[('prod', '<u4'), ('conjugate', 'u1')])
+        dt_prod = np.dtype([('input_a', '<u2'), ('input_b', '<u2')])
+        bt_prod = np.array(np.triu_indices(self.telescope.nfeed)).astype('<u2').T.copy().view(dt_prod).reshape(-1)
```

Comment:
couldn't this also be precalculated in `setup`? It's probably not wasting much time, but it would be a bit more consistent.

Reply:
Good point. I can do that I guess.